core/src/main/scala/com.snowplowanalytics.snowplow.collectors.scalastream/telemetry/TelemetryAkkaService.scala

```diff
@@ -0,0 +1,123 @@
+package com.snowplowanalytics.snowplow.collectors.scalastream.telemetry
+
+import cats.data.NonEmptyList
+
+import akka.actor.ActorSystem
+
+import io.circe.Json
+
+import scala.concurrent.ExecutionContextExecutor
+import scala.concurrent.duration.Duration
+
+import org.slf4j.{Logger, LoggerFactory}
+
+import java.util.concurrent.TimeUnit
+
+import com.snowplowanalytics.iglu.core.SelfDescribingData
+import com.snowplowanalytics.snowplow.scalatracker.Emitter.EndpointParams
+import com.snowplowanalytics.snowplow.scalatracker.Tracker
+import com.snowplowanalytics.snowplow.scalatracker.idimplicits._
+import com.snowplowanalytics.snowplow.scalatracker.emitters.id.SyncEmitter
+import com.snowplowanalytics.snowplow.scalatracker.Emitter._
+import com.snowplowanalytics.snowplow.collectors.scalastream.model.{
+  CollectorConfig,
+  GooglePubSub,
+  Kinesis,
+  Sqs,
+  TelemetryConfig
+}
+
+/** Akka implementation of telemetry service.
+  *
+  * @param teleCfg - telemetry configuration
+  * @param cloud - cloud vendor
+  * @param region - deployment region
+  * @param appName - application name as defined during build (take it from BuildInfo)
+  * @param appVersion - application name as defined during build (take it from BuildInfo)
+  */
+case class TelemetryAkkaService(
+  teleCfg: TelemetryConfig,
+  cloud: Option[CloudVendor],
+  region: Option[String],
+  appName: String,
+  appVersion: String
+) {
+  private lazy val log: Logger = LoggerFactory.getLogger(getClass)
+
+  private lazy val payload: SelfDescribingData[Json] = makeHeartbeatEvent(teleCfg, cloud, region, appName, appVersion)
+
+  def start()(implicit actorSystem: ActorSystem): Unit =
+    if (teleCfg.isDisabled) {
+      log.info(s"Telemetry disabled")
+    } else {
+      log.info(s"Telemetry enabled")
+      val scheduler                                   = actorSystem.scheduler
+      implicit val executor: ExecutionContextExecutor = actorSystem.dispatcher
+
+      def emitterCallback(params: EndpointParams, req: Request, res: Result): Unit =
+        res match {
+          case Result.Success(_) => log.debug(s"telemetry send successfully")
+          case Result.Failure(code) =>
+            log.warn(s"Scala telemetry tracker got unexpected HTTP code $code from ${params.getUri}")
+          case Result.TrackerFailure(exception) =>
+            log.warn(
+              s"Scala telemetry tracker failed to reach ${params.getUri} with following exception $exception" +
+                s" after ${req.attempt} attempt"
+            )
+          case Result.RetriesExceeded(failure) =>
+            log.warn(s"Scala telemetry tracker has stopped trying to deliver payload after following failure: $failure")
+        }
+
+      val emitter: SyncEmitter = SyncEmitter(
+        EndpointParams(teleCfg.url, port = teleCfg.port, https = teleCfg.isSecure),
+        callback = Some(emitterCallback)
```

Comment:
Would be good to perform some retries, see [here](https://github.com/snowplow/enrich/blob/feature/file-sink-rotate/modules/common-fs2/src/main/scala/com/snowplowanalytics/snowplow/enrich/common/fs2/Telemetry.scala#L76)

Reply:
Defaults to 10 retries, same as the example.

https://github.com/snowplow/snowplow-scala-tracker/blob/af50b940cb6c258f294c2bb878b633a0f42c7228/modules/id-emitter/src/main/scala/com/snowplowanalytics/snowplow/scalatracker/emitters/id/SyncEmitter.scala#L89

Reply:
Good point I missed that